apps/storefront/src/components/layout/B3CompanyHierarchy.test.tsx

```diff
@@ -0,0 +1,128 @@
+import { buildCompanyStateWith } from 'tests/storeStateBuilders/companyStateBuilder';
+import { renderWithProviders, screen } from 'tests/test-utils';
+
+import B3CompanyHierarchy from './B3CompanyHierarchy';
+
+describe('B3CompanyHierarchy', () => {
+  describe('when company hierarchy representation is active', () => {
+    it('should show company hierarchy dropdown with representing company chip', () => {
+      const preloadedState = {
+        company: buildCompanyStateWith({
+          companyHierarchyInfo: {
+            selectCompanyHierarchyId: '5',
+            isEnabledCompanyHierarchy: true,
+            isHasCurrentPagePermission: true,
+            companyHierarchyList: [
+              { companyId: 1, companyName: 'Parent Company', channelFlag: false },
+              { companyId: 5, companyName: 'Child Company', channelFlag: false },
+            ],
+            companyHierarchyAllList: [],
+            companyHierarchySelectSubsidiariesList: [],
+          },
+          companyInfo: {
+            id: '1',
+            companyName: 'Parent Company',
+          },
+          pagesSubsidiariesPermission: {
+            order: true,
+            invoice: false,
+            addresses: false,
+            userManagement: false,
+            shoppingLists: false,
+          },
+        }),
+      };
+
+      renderWithProviders(<B3CompanyHierarchy />, {
+        preloadedState,
+        initialGlobalContext: {},
+      });
+
+      const dropdownButton = screen.getByRole('button');
+      expect(dropdownButton).toBeInTheDocument();
+
+      expect(dropdownButton).toHaveTextContent('Child Company');
+
+      expect(screen.getByText('Representing')).toBeInTheDocument();
+    });
+  });
+
+  describe('when no company hierarchy representation is active', () => {
+    it('should show current company chip only', () => {
+      const preloadedState = {
+        company: buildCompanyStateWith({
+          companyHierarchyInfo: {
+            selectCompanyHierarchyId: '',
+            isEnabledCompanyHierarchy: true,
+            isHasCurrentPagePermission: true,
+            companyHierarchyList: [
+              { companyId: 1, companyName: 'Parent Company', channelFlag: false },
+            ],
+            companyHierarchyAllList: [],
+            companyHierarchySelectSubsidiariesList: [],
+          },
+          companyInfo: {
+            id: '1',
+            companyName: 'Parent Company',
+          },
+          pagesSubsidiariesPermission: {
+            order: true,
+            invoice: false,
+            addresses: false,
+            userManagement: false,
+            shoppingLists: false,
+          },
+        }),
+      };
+
+      renderWithProviders(<B3CompanyHierarchy />, {
+        preloadedState,
+        initialGlobalContext: {},
+      });
+
+      const dropdownButton = screen.getByRole('button');
+      expect(dropdownButton).toBeInTheDocument();
+
+      expect(dropdownButton).toHaveTextContent('Parent Company');
```

Comment:
```tsx
const dropdownButton = screen.getByRole('button', { name: 'Parent Company' });
expect(dropdownButton).toBeInTheDocument();
```

if that doesn't work, then simply omit the `toBeInTheDocument`:


```tsx
const dropdownButton = screen.getByRole('button');

expect(dropdownButton).toHaveTextContent('Parent Company');
```

`getBy*` will blow up if the element is not in the document, so there's no need to ensure it is in. We add the `.toBeInTheDocument` when we don't need to make any other check on the element to avoid leaving the variable "unused".